pageserver/src/layered_repository/inmemory_layer.rs

```diff
@@ -308,23 +400,27 @@ impl Layer for InMemoryLayer {
             .unwrap_or_default();
 
         println!(
-            "----- in-memory layer for tli {} seg {} {}-{} {} ----",
-            self.timelineid, self.seg, self.start_lsn, end_str, inner.dropped,
+            "----- in-memory layer for tli {} LSNs {}-{} ----",
+            self.timelineid,
+            self.start_lsn,
+            end_str,
+            //inner.dropped,
         );
+        /*
+                for (k, v) in inner.seg_sizes.iter() {
+                    println!("seg_sizes {} {}: {:?}", k.0, k.1, v);
+                }
 
-        for (k, v) in inner.seg_sizes.as_slice() {
-            println!("seg_sizes {}: {}", k, v);
-        }
-
-        for (blknum, lsn, pos) in inner.page_versions.ordered_page_version_iter(None) {
-            let pv = inner.page_versions.read_pv(pos)?;
-            let pv_description = match pv {
-                PageVersion::Page(_img) => "page",
-                PageVersion::Wal(_rec) => "wal",
-            };
+                for (blknum, lsn, pos) in inner.page_versions.ordered_page_version_iter(None) {
+                    let pv = inner.page_versions.read_pv(pos)?;
+                    let pv_description = match pv {
+                        PageVersion::Page(_img) => "page",
+                        PageVersion::Wal(_rec) => "wal",
+                    };
 
-            println!("blk {} at {}: {}\n", blknum, lsn, pv_description);
-        }
+                    println!("blk {} at {}: {}\n", blknum, lsn, pv_description);
+                }
+        */
```

Comment:
I broke the dump() function and just commented it out instead of fixing it. It's just for debugging purposes and isn't actually used anywhere. The dump() functions for Delta and Imagelayers are used by the 'dump_layerfile' utility, but the InMemoryLayer::dump function is completely unused. It can come handy while debugging though. In any case, fixed it now.